shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -517,8 +517,20 @@ private CharSequence getClipboardData(PlatformChannel.ClipboardContentFormat for
       if (clip == null) return null;
       if (format == null || format == PlatformChannel.ClipboardContentFormat.PLAIN_TEXT) {
         ClipData.Item item = clip.getItemAt(0);
-        if (item.getUri() != null)
-          activity.getContentResolver().openTypedAssetFileDescriptor(item.getUri(), "text/*", null);
+        // First, try getting clipboard as text.
+        CharSequence itemText = item.getText();
+        if (itemText == null) {
+          // Clipboard does not contain text, so check whether or not we will be
+          // able to retrieve text from URI. FileNotFoundException will be thrown
+          // if not, and then null will be returned.
+          if (item.getUri() != null) {
```

Comment:
Even with the restructuring, shouldn't we be checking that the URI is non-null and also `content:`? We know from the docs that it's invalid to call `openTypedAssetFileDescriptor` and we know from the issue that calling this with at least `file:` will throw an exception that can potentially bring down the whole app.

Hopefully moving `getText` up will avoid that in most cases, but we've had a lot of weird plugin bugs over the years that are related to other apps sending us bad data (e.g., intent responses that have unexpected nulls), so it seems like we should have a defense-in-depth approach here to make sure that if someone sends us a `file:` URI with no text it won't throw.

Reply:
I think that explicitly returning null in the case where the URI is null makes sense to make sure no weird errors come up like you're referencing.

In terms of adding a check for `content:` before calling `openTypedAssetFileDescriptor`, that makes sense as well; I initially assumed a `FileNotFoundException` would be thrown (which we catch) which does not seem to be the case.

I think it would also be helpful to add some logs to these new branches to make it clear why null is returned.

Reply:
Addressed these comments, would you be willing to give it a second look? @stuartmorgan